### PR TITLE
[Subscription RP] New-AzureRmSubscription

### PR DESCRIPTION
**Pull request by wilcobmsft**
## Description

Adding a cmdlet New-AzureRmSubscription which will allow EA customers to automate subscription creation.

## Checklist

- [X] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [X] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [X] For public API changes to cmdlets:
    - [X] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
      See [design PR](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/25)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by wilcobmsft** [comment]
Note: the actual [SDK PR](https://github.com/Azure/azure-sdk-for-net/pull/4157) is still open, so checks are not expected to pass yet.

**Comment by wilcobmsft** [comment]
.NET SDK merged.

File: src/ResourceManager/Subscription/Commands.Subscription/Commands.Subscription.Netcore.csproj [hunk 1]
```diff
@@ -39,7 +39,9 @@
 
   <ItemGroup>
     <ProjectReference Include="..\..\..\Common\Commands.Common.Authentication.Abstractions\Common.Authentication.Abstractions.Netcore.csproj" />
+    <ProjectReference Include="..\..\..\Common\Commands.Common.Graph.RBAC\Common.Rbac.Netcore.csproj" />
     <ProjectReference Include="..\..\..\Common\Commands.Common\Common.Netcore.csproj" />
+    <ProjectReference Include="..\..\Common\Commands.Common.Authentication.ResourceManager\Common.ResourceManager.Authentication.Netcore.csproj" />
```
Comment (markcowl) [inline src/ResourceManager/Subscription/Commands.Subscription/Commands.Subscription.Netcore.csproj]: Unfortunately, you cannot take a reference to this project, as it contains dependencies that are controlled by Profile, and may change.  I assume that this is to get the PSAzureSubscription model?

File: src/ResourceManager/Subscription/Commands.Subscription/Commands.Subscription.csproj [hunk 2]
```diff
@@ -44,28 +44,33 @@
...
     </ProjectReference>
+    <ProjectReference Include="..\..\..\Common\Commands.Common.Graph.RBAC\Commands.Common.Graph.RBAC.csproj">
+      <Project>{269acf73-0a34-42dc-ab9c-4b15931a489d}</Project>
+      <Name>Commands.Common.Graph.RBAC</Name>
+    </ProjectReference>
     <ProjectReference Include="..\..\..\Common\Commands.Common\Commands.Common.csproj">
       <Project>{5ee72c53-1720-4309-b54b-5fb79703195f}</Project>
       <Name>Commands.Common</Name>
     </ProjectReference>
+    <ProjectReference Include="..\..\Common\Commands.Common.Authentication.ResourceManager\Commands.Common.Authentication.ResourceManager.csproj">
```
Comment (markcowl) [inline src/ResourceManager/Subscription/Commands.Subscription/Commands.Subscription.csproj]: SHould not take a reference to this (see above)

File: src/ResourceManager/Subscription/Commands.Subscription/Cmdlets/GetAzureRmSubscriptionDefinition.cs [hunk 3]
```diff
@@ -1,43 +0,0 @@
...
-using Microsoft.Azure.Commands.Subscription.Common;
-using Microsoft.Azure.Commands.Subscription.Models;
-using Microsoft.Azure.Management.Subscription;
-using System.Collections.Generic;
-using System.Management.Automation;
-
-namespace Microsoft.Azure.Commands.Subscription.Cmdlets
-{
-    [Cmdlet(VerbsCommon.Get, "AzureRmSubscriptionDefinition", DefaultParameterSetName = ParameterSetNames.Default), OutputType(typeof(List<PSSubscriptionDefinition>))]
-    public class GetAzureRmSubscriptionDefinition : AzureSubscriptionDefinitionCmdletBase
```
Comment (markcowl) [inline src/ResourceManager/Subscription/Commands.Subscription/Cmdlets/GetAzureRmSubscriptionDefinition.cs]: Now that there is no 'Get' cmdlet, how do I get the subscriptions?  Will Get-AzureRmSubscription work?

File: src/ResourceManager/Subscription/Commands.Subscription/Commands.Subscription.Netcore.csproj (hunk 1 above)
Comment (markcowl) [reply src/ResourceManager/Subscription/Commands.Subscription/Commands.Subscription.Netcore.csproj]: If it is for this, we can move this type into another assembly (for example, the common resource manager assembly).

File: src/ResourceManager/Subscription/Commands.Subscription/Cmdlets/GetAzureRmSubscriptionDefinition.cs (hunk 3 above)
Comment (wilcobmsft) [reply src/ResourceManager/Subscription/Commands.Subscription/Cmdlets/GetAzureRmSubscriptionDefinition.cs]: Yes, the existing Get-AzureRmSubscription cmdlet will work.

File: src/ResourceManager/Subscription/Commands.Subscription/Commands.Subscription.Netcore.csproj (hunk 1 above)
Comment (wilcobmsft) [reply src/ResourceManager/Subscription/Commands.Subscription/Commands.Subscription.Netcore.csproj]: Yes, it is just for PSAzureSubscription. I'll just create a new IAzureSubscription implementation instead for now and remove the dependency.

File: src/ResourceManager/Subscription/Commands.Subscription/Commands.Subscription.csproj (hunk 2 above)
Comment (wilcobmsft) [reply src/ResourceManager/Subscription/Commands.Subscription/Commands.Subscription.csproj]: Fixed

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/722/

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/736/

**Comment by msilinda** [comment]
Is there a recommended completed PowerShell script in GitHub to apply Azure RBAC roles into these new subscriptions once created from this?

**Pull request merged**